Jenkinsfile

```diff
@@ -740,7 +740,7 @@ pipeline {
                         }
                     }
                 } // stage('Functional on CentOS 7')
-                stage('Functional on CentOS 8') {
+                stage('Functional on CentOS 7 with Valgrind') {
```

Comment:
You are renaming/repurposing (deleting for all intents and purposes) an existing CentOS 8 testing stage.

Also, per [CORCI-1081](https://jira.hpdd.intel.com/browse/CORCI-1081) you will have to prevent this stage from running on PRs the same way that daos-stack/pipeline-lib#227 is doing.

Reply:
> You are renaming/repurposing (deleting for all intents and purposes) an existing CentOS 8 testing stage.

d6fb950f697d6ac135b7af6c9d55a8d0f9338f9f

> Also, per CORCI-1081 you will have to prevent this stage from running on PRs the same way that daos-stack/pipeline-lib#227 is doing.

K, so when daos-stack/pipeline-lib#227 lands, I'll do something like this?

```diff
--- a/vars/skipStage.groovy
+++ b/vars/skipStage.groovy
@@ -22,8 +22,8 @@ boolean skip_ftest(String distro) {
            skip_stage_pragma('func-test-' + distro)
 }

-boolean skip_ftest_valgrind(String distro) {
-    return skip_ftest(distro) ||
+boolean skip_ftest_valgrind(String distro, String target_branch) {
+    return skip_ftest(distro, target_branch) ||
            skip_stage_pragma('func-test-vm-valgrind')
 }

@@ -176,7 +176,7 @@ boolean call(Map config = [:]) {
         case "Functional on CentOS 7":
             return skip_ftest('el7')
         case "Functional on CentOS 7 with Valgrind":
-            return skip_ftest_valgrind('el7')
+            return skip_ftest_valgrind('el7', target_branch)
         case "Functional on CentOS 8":
             return skip_ftest('el8')
         case "Functional on Leap 15":
```

Reply:
It's a bit difficult to say without seeing it all together, but you are probably going to need one of these https://github.com/daos-stack/pipeline-lib/blob/a5a5ea52706f1b5e0db38caa37925a62af55904c/vars/skipStage.groovy#L29-L32 to allow a PR to force the enablement of your stage which will be skipped by default.

Then you are probably going to need to extend https://github.com/daos-stack/pipeline-lib/blob/a5a5ea52706f1b5e0db38caa37925a62af55904c/vars/skipStage.groovy#L40 so that your new EL7 stage does not get run by default.

In spelling it out, it seems pretty messy.  :frowning: Maybe once it's all together in a PR, we might see a more elegant way.

Reply:
[CORCI-1081](https://jira.hpdd.intel.com/browse/CORCI-1081) / daos-stack/pipeline-lib#227 has landed, so you will need to update pipeline-lib to skip your new stage in this PR before this can be approved and land.

Reply:
https://github.com/daos-stack/pipeline-lib/pull/216/commits/027ee8f28079b445d9b9751b2352be6116fb3425